File: 2784-check-if-array-is-good/2784-check-if-array-is-good.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        n = len(nums)
        freq = [0] * n
        for i in range(n):
            if nums[i] >= n or nums[i] <= 0:
                return False
            freq[nums[i]]+=1
            if freq[nums[i]] >= 3:
                return False
            elif freq[nums[i]] >= 2 and nums[i] != n - 1:
                return False
        return True
            

File: 2784-check-if-array-is-good/2784-check-if-array-is-good.cpp
class Solution {
public:
    bool isGood(vector<int>& nums) {
        int n = nums.size();
        vector<int>freq(n, 0);
        for (const int& num: nums) {
            if (num >= n || num <= 0) return false;
            freq[num]++;
            if (freq[num] > 1 && num != n - 1) return false;
            if (freq[num] > 2 && num == n - 1) return false;
        }
        return true;
    }
};